app/controllers/newflow/login_signup_controller.rb

```diff
@@ -0,0 +1,323 @@
+module Newflow
+  # Contains every action for login and signup
+  class LoginSignupController < ApplicationController
+    layout 'newflow_layout'
+    skip_before_action :authenticate_user!
+    before_action :newflow_authenticate_user!, only: [:profile_newflow]
+    before_action :save_new_params_in_session, only: [:login_form]
+    before_action :maybe_skip_to_sign_up, only: [:login_form]
+    before_action :store_authorization_url_as_fallback, only: [:login_form, :login, :signup_form, :signup]
+    before_action :known_signup_role_redirect, only: [:login_form]
+    before_action :restart_if_missing_unverified_user, only: [:verify_email, :verify_pin, :change_your_email]
+    before_action :exit_newflow_signup_if_logged_in, only: [:login_form, :signup_form, :student_signup_form, :welcome]
+    before_action :set_active_banners
+    before_action :cache_client_app, only: [:login, :welcome]
+    skip_before_action :check_if_password_expired
+    fine_print_skip :general_terms_of_use, :privacy_policy,
+                    except: [:profile_newflow, :verify_pin, :signup_done]
```

Comment:
Wait, those two sentences sounds contradictory to me. I'm going with your first recommendation because it makes the code cleaner (even though it seems to imply that there is an order to them—which there isn't, not really??).